en/development/testing.rst

```diff
@@ -1322,6 +1322,281 @@ comparison files as they are referenced:
+
+            $parser
+                ->setDescription('My cool console app')
+                ->addSubcommand('updateModified', [
+                    'parser' => $updateModifiedParser
+                ])
+                // add a new subcommand
+                ->addSubcommand('bestFramework');
+            return $parser;
+        }
+
+        public function updateModified()
+        {
+            $table = $this->args[0];
+            $this->loadModel($table);
+            $this->{$table}->query()
+                ->update()
+                ->set([
+                    'modified' => new FrozenTime()
+                ])
+                ->execute();
+        }
+
+        // create this interactive subcommand
+        public function bestFramework()
+        {
+            $this->out('Hi there!');
+
+            $framework = $this->in('What is the best PHP framework?');
+            if ($framework !== 'CakePHP') {
+                $this->err("I disagree that '$framework' is the best.");
+                $this->_stop(Shell::CODE_ERROR);
+            }
+
+            $this->out('I agree!');
+        }
+    }
+
+Now that we have an interactive subcommand, we can add a test case that tests
+that we receive the proper response, and one that tests that we receive an
+incorrect response. Add the following methods to
+**tests/TestCase/Shell/MyConsoleShellTest.php**::
+
+    public function testBestFramework()
+    {
+        $this->exec('my_console best_framework', [
+            'CakePHP'
+        ]);
+        $this->assertExitCode(Shell::CODE_SUCCESS);
+        $this->assertOutputContains('I agree!');
+    }
+
+    public function testBestFrameworkWrongAnswer()
+    {
+        $this->exec('my_console best_framework', [
+            'my homemade framework'
+        ]);
+        $this->assertExitCode(Shell::CODE_ERROR);
+        $this->assertErrorRegExp("/I disagree that \'(.+)\' is the best\./");
+    }
+
+As you can see from the ``testBestFramework``, it responds to the first input
+request with "CakePHP". Since this is the correct answer according to our
+subcommand, the shell will exit successfully after outputting a response.
+
+The second test case, ``testBestFrameworkWrongAnswer``, provides an incorrect
+answer which causes our shell to fail and exit with ``1``. We also assert
+that ``stderr`` was given our error, which includes the name of the incorrect
+answer.
+
+Testing the CommandRunner
+-------------------------
+
+To test shells that are dispatched using the ``CommandRunner`` class, enable it
+in your test case with the following method::
+
+    $this->useCommandRunner();
+
+.. versionadded:: 3.5.0
+
+    The ``CommandRunner`` class was added.
+
+Assertion methods
+-----------------
+
+The ``Cake\TestSuite\ConsoleIntegrationTestCase`` class provides a number of
+assertion methods that make it easy to assert against console output::
+
+    // assert that the shell exited with the expected code
+    $this->assertExitCode($expected);
```

Comment:
I'm okay with that, but probably `assetExitCodeSuccess` so it more closely resembles `assertResponseOk`, `assertResponseError`, etc.